src/app/components/Search/TableSearchBar.tsx

```diff
@@ -0,0 +1,133 @@
+import SearchIcon from '@mui/icons-material/Search'
+import HighlightOffIcon from '@mui/icons-material/HighlightOff'
+import InputAdornment from '@mui/material/InputAdornment'
+import { COLORS } from '../../../styles/theme/colors'
+import IconButton from '@mui/material/IconButton'
+import { useScreenSize } from '../../hooks/useScreensize'
+import WarningIcon from '@mui/icons-material/WarningAmber'
+import { typingDelay } from '../../../styles/theme'
+import Typography from '@mui/material/Typography'
+import { useTranslation } from 'react-i18next'
+import Button from '@mui/material/Button'
+import { CardEmptyState } from '../CardEmptyState'
+import { inputBaseClasses } from '@mui/material/InputBase'
+
+export interface TableSearchBarProps {
+  placeholder: string
+  warning?: string
+  value: string
+  onChange: (value: string) => void
+}
+
+export const TableSearchBar: FC<TableSearchBarProps> = ({ value, onChange, placeholder, warning }) => {
+  const { isTablet } = useScreenSize()
+
+  const [isProblemFresh, setIsProblemFresh] = useState(false)
+
+  useEffect(() => {
+    if (warning) {
+      const timeout = setTimeout(() => {
+        setIsProblemFresh(false)
+      }, typingDelay)
+      return () => clearTimeout(timeout)
+    } else {
+      setIsProblemFresh(true)
+    }
+  }, [warning])
+
+  const startAdornment = (
+    <InputAdornment
+      position="start"
+      disablePointerEvents // Pass clicks through, so it focuses the input
+    >
+      <SearchIcon sx={{ color: COLORS.grayDark, ml: -3 }} />
+    </InputAdornment>
+  )
+
+  const onClearValue = () => onChange('')
+
+  const endAdornment = (
+    <InputAdornment position="end">
+      {value ? (
+        <IconButton color="inherit" sx={{ mt: -3, mb: -3, mr: -1 }} onClick={onClearValue}>
+          <HighlightOffIcon />
+        </IconButton>
+      ) : (
+        <span style={{ width: '38px' }} />
+      )}
+    </InputAdornment>
+  )
+
+  const helperText = isProblemFresh ? undefined : (
+    <Typography
+      component="span"
+      sx={{
+        display: 'inline-flex',
+        color: COLORS.warningColor,
+        fontSize: 12,
+        lineHeight: 2,
+        alignItems: 'center',
+        verticalAlign: 'middle',
+        mt: 3,
+        mb: 4,
+        width: isTablet ? '160px' : undefined,
+      }}
+    >
+      <WarningIcon sx={{ mr: 3 }} />
+      {warning}
+    </Typography>
+  )
+
+  return (
+    <TextField
+      sx={{
+        backgroundColor: COLORS.white,
+        marginLeft: 4,
+        marginRight: helperText ? '25px' : '25px',
+        '&:focus-within': {
+          boxShadow: '3px 3px 3px 3px rgb(0, 0, 98, 0.25) !important',
+        },
+        [`.${inputBaseClasses.root}`]: {
+          border: '1px solid',
+          borderColor: COLORS.inactiveStroke,
+        },
+        ...(helperText
+          ? {
+              border: '1px solid',
+              borderColor: COLORS.inactiveStroke,
+              marginBottom: isTablet ? '-99px' : '-50px',
```

Comment:
Might look weird on devices with different DPI than yours.

Reply:
No, that makes it inconsistent with what we are using(majority units defined in px). But yeah, we should have gone with `em` type units from the start.

Reply:
I mean, converting px -> em, does not belong in this PR. We can bring it up on next FE sync to discuss.

Reply:
OK, I agree.